flyteplugins/go/tasks/pluginmachinery/flytek8s/pod_helper.go

```diff
@@ -38,9 +38,15 @@ const primaryContainerTemplateName = "primary"
 const primaryInitContainerTemplateName = "primary-init"
 const PrimaryContainerKey = "primary_container_name"
 
-// nodePreemptionStatusReasons are the status reasons that a pod's respective node
-// is preempted by the scheduler
-var nodePreemptionStatusReasons = sets.NewString("Shutdown", "Terminated", "NodeShutdown")
+var retryableStatusReasons = sets.NewString(
+	// Reasons that indicate the node was preempted aggressively.
+	// Kubelet can miss deleting the pod prior to the node being shutdown.
+	"Shutdown",
+	"Terminated",
+	"NodeShutdown",
+	// kubelet admission rejects the pod before the node gets assigned appropriate labels.
+	"NodeAffinity",
```

Comment:
Are we sure that the `NodeAffinity` reason is enough to determine if this should be retried? Worse case scenario is that we retry something that should not be (not all that bad), but was to gauge confidence here.

Reply:
Yeah, this was my concern as well. I have more confidence in that the node is marked as Failed with the following

```
status: {
message: "Pod was rejected: Predicate NodeAffinity failed: node(s) didn't match Pod's node affinity/selector"
phase: "Failed"
reason: "NodeAffinity"
startTime: "2025-05-16T04:53:42Z"
}
}
```
.

Given the specific nature of this status, I am assuming a small likelihood that we would have a prevalance of false positives causing us to retry needlessly.